packages/api/tests/models/granules/test-granule-reporting.js

```diff
@@ -253,6 +267,96 @@ test('_storeGranuleRecord() will allow a running status to replace a failed stat
   );
 });
 
+test('_storeGranuleRecord() will allow a completed status to replace a queued status for a new execution', async (t) => {
+  const { granuleModel } = t.context;
+
+  const granule = fakeGranuleFactoryV2({ status: 'queued' });
+
+  await granuleModel._storeGranuleRecord(granule);
+  const updateTime = Date.now();
+
+  const updatedGranule = {
+    ...granule,
+    execution: 'new-execution-url',
+    status: 'completed',
+    createdAt: updateTime,
+  };
+
+  await granuleModel._storeGranuleRecord(updatedGranule);
+
+  const fetchedItem = await granuleModel.get({ granuleId: granule.granuleId });
+
+  t.deepEqual(
+    fetchedItem,
+    {
+      ...granule,
+      status: 'completed',
+      createdAt: updateTime,
+      execution: 'new-execution-url',
+    }
+  );
+});
+
+test('_storeGranuleRecord() will allow a running status to replace a queued status for a new execution', async (t) => {
+  const { granuleModel } = t.context;
+
+  const granule = fakeGranuleFactoryV2({ status: 'queued' });
+
+  await granuleModel._storeGranuleRecord(granule);
+  const updateTime = Date.now();
+
+  const updatedGranule = {
+    ...granule,
+    execution: 'new-execution-url',
+    status: 'running',
+    createdAt: updateTime,
+  };
+
+  await granuleModel._storeGranuleRecord(updatedGranule);
+
+  const fetchedItem = await granuleModel.get({ granuleId: granule.granuleId });
+
+  t.deepEqual(
+    fetchedItem,
+    {
+      ...granule,
+      status: 'running',
+      createdAt: updateTime,
+      execution: 'new-execution-url',
+    }
+  );
+});
+
+test('_storeGranuleRecord() will allow a queued status to replace a running status for a new execution', async (t) => {
```

Comment:
Is there a test for something like `_storeGranuleRecord() will not allow a queued status to replace any other status for same execution`? I guess since we don't know the executionArn now at time of status update to queued, that may still be an issue/hard to determine...

Reply:
I think we also want a test to prove that a queued status cannot replace a running/completed status for the same execution

Reply:
@markdboyd It's fair enough to test, but I am concerned that we won't see that side of things being used in `queue-granules` because it's not sending any execution ID with the status update.

Reply:
Perhaps we should be avoiding the race condition in the update logic and indeed do the update after the call to `enqueue...` so that we have and are able to include the executionArn in the update.

Reply:
Regardless of how we write `queue-granules`, I think we need to be sure that our core database write logic does the correct thing, especially since we now have an endpoint that people can call at any time when the database could be in any state